t/24-worker-engine.t

```diff
@@ -227,13 +229,20 @@ subtest 'symlink testrepo' => sub {
     $result  = OpenQA::Worker::Engines::isotovideo::_link_repo($casedir, $pool_directory, 'opensuse');
     is $result, undef, 'create symlink successfully';
 
-    $settings->{DISTRI}   = 'mine';
-    $settings->{JOBTOKEN} = 'token99916';
-    delete $settings->{NEEDLES_DIR};
-    $settings->{CASEDIR} = 'https://github.com/foo/os-autoinst-distri-example.git#master';
-    $job                 = OpenQA::Worker::Job->new($worker, $client, {id => 12, settings => $settings});
-    $result              = OpenQA::Worker::Engines::isotovideo::engine_workit($job);
+    $settings->{DISTRI}      = 'fedora';
+    $settings->{JOBTOKEN}    = 'token99916';
+    $settings->{NEEDLES_DIR} = 'fedora/needles';
+    $settings->{CASEDIR}     = 'https://github.com/foo/os-autoinst-distri-example.git#master';
+    my $productdir = productdir('fedora', undef, undef);
```

Comment:
hm, not sure if this is still a good test when you use a library function to define a productdir. I would prefer a simple plain-text test data input for testing purposes.

Reply:
`$productdir` is not used for input here as far as I can see. But it might still be good to avoid the function call and hard-code the expected value based on the other inputs to avoid relying on how productdir works.

Reply:
OK. I will change it to hard-code. I wanted to use this function to count the `$productdir` value. And make sure the value is the default one.